test/fixtures/bats/exported_function.bats

```diff
@@ -0,0 +1,10 @@
+# see issue https://github.com/sstephenson/bats/issues/225
+
+if exported_function; then
```

Comment:
What happens if you try `a=$(declare -F exported_function >/dev/null)` and run it with different Bash versions?

Reply:
It works with the following:

```bash
# in exported_function.bats
a=$(declare -F exported_function)
```
```bash
# in bats.bats
exported_function() { :; }
```

I downgraded my Bash to 4.3.48(2), and it gives me this:
```
$ exported_function() { :; }
$ export -f exported_function
$ bin/bats --tap test/fixtures/bats/exported_function.bats 
1..1
not ok 1 failing test
# (in file , line 9,
#  from function `bats_perform_test' in file libexec/bats-exec-test, line 313,
#  from function `main' in test file , line 363)
```

### Trying different approaches (original one with `if exported_function; then` only works in 4.4).

This one works in all versions:
```bash
# in exported_function.bats
a=$(exported_function && echo 'exported_function')
```
```bash
# in bats.bats
exported_function() { true; }
```

---

This one works only in 4.4:
```bash
# in exported_function.bats
exported_function && a='exported_function'
```
```bash
# in bats.bats
exported_function() { true; }
```

---

This one works in all versions:
```bash
# in exported_function.bats
a=$(exported_function)
```
```bash
# in bats.bats
exported_function() { echo exported_function; }
```

---

The last one works only in 4.4:
```bash
# in exported_function.bats
exported_function && a='exported_function'
```
```bash
# in bats.bats
exported_function() { return 0; }
```

---

To me it seems that executing `exported_function` in the current process breaks things up, whereas in a subprocess (like `$(exported_function)`) works properly.

Reply:
Interesting, especially when you consider that `run bats` should be launching a subshell anyways.

Out of curiosity, what happens if you put this `if` condition in `setup()` or inside a `@test` case?

Reply:
If I put it in `setup() { ... }` at the beginning of `exported_function.bats`, it works only in Bash 4.4. If I put it inside the `failing test` test case, same thing, it works only in Bash 4.4.